modules/k8s/pod.go

```diff
@@ -128,11 +141,38 @@ func WaitUntilPodAvailableE(t testing.TestingT, options *KubectlOptions, podName
 			return "Pod is now available", nil
 		},
 	)
+	if err == nil {
+		for i := successes; ; i-- {
+			var pod *corev1.Pod
+			pod, err = GetPodE(t, options, podName)
+			if err != nil {
+				break
+			}
+			if !IsPodAvailable(pod) {
+				err = NewPodNotAvailableError(pod)
+				break
+			}
+			if i == 0 {
+				break
+			}
+			logger.Logf(t, "Wait %s before checking if pod %s is still available", sleepBetweenRetries, podName)
+			time.Sleep(sleepBetweenRetries)
```

Comment:
This logic looks to be for a quite specific case: pod becomes ready and after a while changes status to not ready - not sure if should be included in PR for adding additional logs when pod fail to start

Reply:
Up to you I can either do this as a separate issue or update PR title. The reason why this was added here is that in the case of a failing startup the pod may become ready and very quickly switch to not ready.

Reply:
I think this should go as a separate feature